### DISCUSSION TITLE
RTK Query: Query Overview

### DISCUSSION THREAD

**Discussion by ThomasRutzer** [reduxjs/redux-toolkit#3010] [Q&A] [answered]
Hi, I'd like to know how you guys would handle that: 

I have a search query, which accepts a search term. And in one component consuming this queries, I'd like to have an overview about all the search queries made. The only solution I found right now is to map over all the queries in the redux store and filter by `endpointName`. Is there any more sophisticated way to do so?

**Comment by markerikson** [comment, marked as answer]
That would be the primary way to do it.

Another option might be to have another slice listen to that query endpoint's actions using `api.endpoints.someQueryEndpoint.matchFulfilled`, and collect the cache key values over in the slice for later display.

**Reply by ThomasRutzer** [reply]
Wow, thanks for this quick answer @markerikson, appreciate it a lot. I'll like your alternative answer a little more, going to try that and might post a working draft here for others.

**Reply by ThomasRutzer** [reply]
The only other thing about your other solution: Could the other slice be somehow informed about invalidations concerning ths endpointName?

**Reply by markerikson** [reply]
What do you mean "informed about invalidations"?  Can you give an example?

You might be able to use `.matchPending` to listen for new fetches for that endpoint.

**Reply by ThomasRutzer** [reply]
I mean if I would select the queries from store, than if one query would be invalidated, I understand, this would cause a re-render in the component displaying the query data. But if I just listen in a slice to `matchFulfilled` of each query and store the payload, would there be any way to erase an entry from the slice?

**Reply by markerikson** [reply]
Well, invalidation means re-fetching the data, so not sure why you'd want to erase an entry there.

If you're talking about handling when a cache entry gets _removed_, you could listen for `api.internalActions.removeQueryResult` and do some checks against the action contents to see if it's relevant for this data.

**Reply by ThomasRutzer** [reply]
Ah yes, `removeQueryResult` is what I am looking for. Sorry for pointing in a wrong direction. You`ve been very helpful and thanks for your good work on this tool!
